roles/common/tasks/main.yml

```diff
@@ -10,6 +10,13 @@
   when: ansible_version is not defined or false in [{% for item in ansible_requirements %}{{ ansible_version.full | version_compare(item.version, item.operator) }},{% endfor %}]
   run_once: true
 
+- name: Validate format of site_hosts
+  fail:
+    msg: "{{ lookup('template', 'site_hosts.j2') }}"
+  with_dict: "{{ wordpress_sites }}"
+  when: 0 < [{% for host in item.value.site_hosts if 'canonical' not in host %}'',{% endfor %}] | count
```

Comment:
@swalkinshaw Will use your superior suggestion of 
`when: item.value.site_hosts | rejectattr('canonical', 'defined') | list | count`
Quite beautiful. Thank you.